ftd/src/interpreter/test.rs

```diff
@@ -261,27 +191,26 @@ fn evalexpr_test() {
 
 #[test]
 fn test_extract_kwargs() {
-    let doc = interpret_helper(
+    let doc = ftd::parse_doc(
         "foo",
-        r#"-- component foo:
-    kw-args data:
+        r#"
+            -- component fizz:
+            kw-args data:
 
-    -- ftd.text: Hello world
+            -- ftd.text: Hello world
 
-    -- end: foo
+            -- end: fizz
 
-    -- foo:
-    bar: Hello
-    baz: World
-"#,
+            -- fizz:
+            id: test
+            bar: Hello
+            baz: World
+        "#,
     )
     .unwrap();
 
-    let tdoc = doc.tdoc();
-    let instructions = doc.get_instructions("foo#foo");
-    let instruction = instructions.first().unwrap();
-
-    let data = instruction.get_kwargs("data", &tdoc).unwrap();
+    let component = doc.get_component_by_id("foo#fizz", "test").unwrap();
```

Comment:
I would prefer if first argument was not needed. It should act like dom.get_element_by_id() in HTML (its playing exactly the same role, so why not copy their obviously superior behaviour).